File: Python/cookie_handling.py
import requests
import pickle
import os

LOGIN_URL = "URL"
PROTECTED_URL = "protected_URL"
COOKIE_FILE = "cookies.pkl"

def save_cookies(session, filename):
    with open(filename, "wb") as f:
        pickle.dump(session.cookies, f)

def load_cookies(session, filename):
    with open(filename, "rb") as f:
        session.cookies.update(pickle.load(f))

def login_and_save():
    session = requests.Session()
    payload = {"username": "your_username", "password": "your_password"}
    resp = session.post(LOGIN_URL, data=payload)
    resp.raise_for_status()
    print("Logged in, status:", resp.status_code)

    save_cookies(session, COOKIE_FILE)
    print(f"Cookies saved to {COOKIE_FILE}")

    dashboard = session.get(PROTECTED_URL)
    print("Protected page title snippet:", dashboard.text[:200])

def reuse_cookies():
    session = requests.Session()
    load_cookies(session, COOKIE_FILE)
    print(f"Loaded cookies from {COOKIE_FILE}")

    dashboard = session.get(PROTECTED_URL)
    print("Protected page title snippet:", dashboard.text[:200])

if __name__ == "__main__":
    if not os.path.exists(COOKIE_FILE):
        login_and_save()
    else:
        reuse_cookies()

File: NodeJS/cookie_handling.js
const axios = require('axios');
const { wrapper } = require('axios-cookiejar-support');
const tough = require('tough-cookie');
const fs = require('fs');

const LOGIN_URL = 'URL';
const PROTECTED_URL = 'protected_URL';
const COOKIE_FILE = 'cookies.json';

const jar = new tough.CookieJar();
const client = wrapper(axios.create({ jar, withCredentials: true }));

async function saveCookies() {
  const serialized = await jar.serialize();
  fs.writeFileSync(COOKIE_FILE, JSON.stringify(serialized));
  console.log(`Cookies saved to ${COOKIE_FILE}`);
}

async function loadCookies() {
  if (!fs.existsSync(COOKIE_FILE)) return false;
  const data = JSON.parse(fs.readFileSync(COOKIE_FILE));
  await jar.deserialize(data);
  console.log(`Loaded cookies from ${COOKIE_FILE}`);
  return true;
}

async function loginAndSave() {
  const resp = await client.post(LOGIN_URL, {
    username: 'your_username',
    password: 'your_password'
  });
  console.log('Logged in, status:', resp.status);

  await saveCookies();

  const dash = await client.get(PROTECTED_URL);
  console.log('Protected page snippet:', dash.data.slice(0, 200));
}

async function reuseCookies() {
  const loaded = await loadCookies();
  if (!loaded) {
    console.log('No cookies found, please run login first.');
    return;
  }
  const dash = await client.get(PROTECTED_URL);
  console.log('Protected page snippet:', dash.data.slice(0, 200));
}

(async () => {
  if (!fs.existsSync(COOKIE_FILE)) {
    await loginAndSave();
  } else {
    await reuseCookies();
  }
})();
